libs/components/src/lib/video-player/video-player.ts

```diff
@@ -0,0 +1,237 @@
+import { applyMixins, FoundationElement } from '@microsoft/fast-foundation';
+import { attr } from '@microsoft/fast-element';
+import videojs from 'video.js';
+import { MediaSkipBy } from '../enums';
+import { Localized } from '../../shared/patterns';
+
+export const DEFAULT_PLAYBACK_RATES = '0.5, 1, 1.5, 2';
+
+function getPlaybackRatesArray(playbackRates: string): number[] {
+	if (playbackRates === '') return [];
+	const ratesArray: number[] = [];
+	
+	playbackRates.split(',').forEach((numStr: string) => {
+		const num = Number(numStr);
+		if (!isNaN(num)) ratesArray.push(num);
+	});
+	return ratesArray;
+}
+
+/**
+ * Base class for video-player
+ *
+ * @public
+ * @slot - Default slot
+ * @event play - Fired when the video is played
+ * @event pause - Fired when the video is paused
+ * @event ended - Fired when the video is ended
+ */
+export class VideoPlayer extends FoundationElement {
+	/**
+	 * Reference to an image which is displayed before the video is played
+	 *
+	 * @public
+	 * @remarks
+	 * HTML Attribute: poster
+	 */
+	@attr poster?: string;
+
+	/**
+	 * URL of a video file
+	 *
+	 * @public
+	 * @remarks
+	 * HTML Attribute: src
+	 */
+	@attr src?: string;
+
+	/**
+	 * @internal
+	 */
+	srcChanged() {
+		this.#initVideo();
+	}
+
+	/**
+	 * Allows the video will play automatically (muted)
+	 *
+	 * @public
+	 * @remarks
+	 * HTML Attribute: autoplay
+	 */
+	@attr({ mode: 'boolean' }) autoplay = false;
+
+	/**
+	 * @internal
+	 */
+	autoplayChanged() {
+		if (this.player) {
+			this.#initVideo();
+		}
+	}
+
+	/**
+	 * Allows the video to loop back to the beginning when finished
+	 *
+	 * @public
+	 * @remarks
+	 * HTML Attribute: loop
+	 */
+	@attr({ mode: 'boolean' }) loop = false;
+
+	/**
+	 * Sets the available playback rates. When an empty string, no choices will be available
+	 *
+	 * @public
+	 * @remarks
+	 * HTML Attribute: playback-rates
+	 */
+	@attr({attribute: 'playback-rates', mode: 'fromView'}) playbackRates: string = DEFAULT_PLAYBACK_RATES;
+
+	/**
+	 * Allows the video to loop back to the beginning when finished
+	 *
+	 * @public
+	 * @remarks
+	 * HTML Attribute: skip-by
+	 */
+	@attr({attribute: 'skip-by', mode: 'fromView'}) skipBy: MediaSkipBy = MediaSkipBy.Ten;
+
+	/**
+	 * @internal
+	 */
+	skipByChanged() {
+		if (this.player) {
+			this.#initVideo();
+		}
+	}
+
+	/**
+	 * @internal
+	 */
+	player: any;
```

Comment:
We can prefix it at least:
```suggestion
	_player: any;
```

Reply:
We seem to have conflicting views on code conventions as Yonatan requested the prefix should be removed.